pymacaroons/serializers/binary_serializer.py

```diff
@@ -1,19 +1,44 @@
 from __future__ import unicode_literals
 
 import binascii
+from collections import namedtuple
+import six
 import struct
-from base64 import urlsafe_b64encode, urlsafe_b64decode
+import sys
+from base64 import urlsafe_b64encode
 
-from pymacaroons.utils import convert_to_bytes
+from pymacaroons.utils import (
+    convert_to_bytes,
+    convert_to_string,
+    raw_b64decode,
+)
 from pymacaroons.serializers.base_serializer import BaseSerializer
 from pymacaroons.exceptions import MacaroonSerializationException
 
+_LOCATION = 1
+_IDENTIFIER = 2
+_VID = 4
+_SIGNATURE = 6
+_EOS = 0
+
+PacketV2 = namedtuple('PacketV2', ['field_type', 'data'])
+
 
 class BinarySerializer(BaseSerializer):
 
     PACKET_PREFIX_LENGTH = 4
 
     def serialize(self, macaroon):
+        return urlsafe_b64encode(
+            self.serialize_raw(macaroon)).decode('ascii').rstrip('=')
+
+    def serialize_raw(self, macaroon):
```

Comment:
For deserialization it's still likely that you will know what type expect ahead of time. If not, we could keep a `BinarySerializer` that delegates to the correct v1/v2. This is fine for now, though.